cover_agent/UnitTestGenerator.py

```diff
@@ -403,6 +470,19 @@ def validate_test(self, generated_test: dict, generated_tests_dict: dict):
 
                 # Step 3: Check for pass/fail from the Runner object
                 if exit_code != 0:
+                    # Test failed due to compilation error , roll back the test file to it's original content
+                    if "syntax error" in stderr or "SyntaxError" in stderr or "IndentationError" in stderr or "ImportError" in stderr:
+                        with open(self.test_file_path, "w") as test_file:
+                            test_file.write(original_content)
+                        self.logger.info(f"Skipping a generated test that failed due to compilation error")
+                        return {
+                            "status": "COMPILATION_ERROR",
+                            "reason": "Compiltaion error",
+                            "exit_code": exit_code,
+                            "stderr": stderr,
+                            "stdout": stdout,
+                            "test": generated_test,
+                        }
```

Comment:
why do you need these lines, and how they are aligned with the logic that already exist?